test/test_library/unit/test_database.py

```diff
@@ -0,0 +1,971 @@
+        TableFactory(dataset=ds1)
+        self.sqlite_db.session.commit()
+
+    def test_creates_column_table(self):
+
+        # Now all tables are created. Can we use ORM to create columns?
+
+        # Column requires table and dataset.
+        ds1 = DatasetFactory()
+        self.sqlite_db.session.commit()
+
+        table1 = TableFactory(dataset=ds1)
+        self.sqlite_db.session.commit()
+
+        ColumnFactory(table=table1)
+        self.sqlite_db.session.commit()
+
+    def test_creates_file_table(self):
+        FileFactory()
+        self.sqlite_db.session.commit()
+
+    def test_creates_partition_table(self):
+
+        ds1 = DatasetFactory()
+        PartitionFactory(dataset=ds1)
+        self.sqlite_db.session.commit()
+
+    def test_creates_columnstat_table(self):
+        self.sqlite_db.session.commit()
+        ColumnStatFactory()
+        self.sqlite_db.session.commit()
+
+    # ._add_config_root
+    def test_creates_new_root_config(self):
+        # prepare state
+        datasets = self.query(Dataset).all()
+        self.assertEquals(len(datasets), 0)
+
+        # testing
+        self.sqlite_db._add_config_root()
+        datasets = self.query(Dataset).all()
+        self.assertEquals(len(datasets), 1)
+        self.assertEquals(datasets[0].name, ROOT_CONFIG_NAME)
+        self.assertEquals(datasets[0].vname, ROOT_CONFIG_NAME_V)
+
+    def test_closes_session_if_root_config_exists(self):
+        # first assert signatures of the functions we are going to mock did not change.
+        assert_spec(self.sqlite_db.close_session, ['self'])
+
+        # prepare state
+        DatasetFactory(id_=ROOT_CONFIG_NAME, vid=ROOT_CONFIG_NAME)
+        self.sqlite_db.session.commit()
+        self.sqlite_db.close_session = fudge.Fake('close_session').expects_call()
+
+        # testing
+        self.sqlite_db._add_config_root()
+        fudge.verify()
+
+    # ._clean_config_root tests
+    def tests_resets_instance_fields(self):
+        ds = DatasetFactory()
+        ds.id_ = ROOT_CONFIG_NAME
+        ds.name = 'name'
+        ds.vname = 'vname'
+        ds.source = 'source'
+        ds.dataset = 'dataset'
+        ds.creator = 'creator'
+        ds.revision = 33
+        self.sqlite_db.session.merge(ds)
+        self.sqlite_db.commit()
+
+        self.sqlite_db._clean_config_root()
+
+        # refresh dataset
+        ds = self.query(Dataset).filter(
+            Dataset.id_ == ROOT_CONFIG_NAME).one()
+        self.assertEquals(ds.name, ROOT_CONFIG_NAME)
+        self.assertEquals(ds.vname, ROOT_CONFIG_NAME_V)
+        self.assertEquals(ds.source, ROOT_CONFIG_NAME)
+        self.assertEquals(ds.dataset, ROOT_CONFIG_NAME)
+        self.assertEquals(ds.creator, ROOT_CONFIG_NAME)
+        self.assertEquals(ds.revision, 1)
+
+    # .inserter test
+    # TODO: ValueInserter does not work without bundle. Fix it.
+    @unittest.skip('ValueInserter requires bundle, but inserter method gives None instead.')
```

Comment:
OK to skip this one. Inserters should get a functional test outside of the bundle.